home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/InvitationApiImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.softserveinc.ita.homeproject.application.api;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+import com.softserveinc.ita.homeproject.api.InvitationsApi;
+import com.softserveinc.ita.homeproject.homeservice.dto.InvitationDto;
+import com.softserveinc.ita.homeproject.homeservice.service.ApartmentInvitationService;
+import com.softserveinc.ita.homeproject.homeservice.service.CooperationInvitationService;
+import com.softserveinc.ita.homeproject.model.InvitationStatus;
+import com.softserveinc.ita.homeproject.model.InvitationToken;
+import com.softserveinc.ita.homeproject.model.ReadApartmentInvitation;
+import com.softserveinc.ita.homeproject.model.ReadCooperationInvitation;
+import com.softserveinc.ita.homeproject.model.ReadInvitation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+
+@Provider
+@Component
+public class InvitationApiImpl extends CommonApi implements InvitationsApi {
+
+    @Autowired
+    ApartmentInvitationService apartmentInvitationService;
+
+    @Autowired
+    CooperationInvitationService cooperationInvitationService;
+
+    @Override
+    public Response approveInvitation(InvitationToken token){
+        ReadInvitation readInvitation = registerWithCorrectService(token.getInvitationToken());
+        return Response.status(Response.Status.CREATED).entity(readInvitation).build();
+    }
+
+    private ReadInvitation registerWithCorrectService(String token){
+        InvitationDto invitation = apartmentInvitationService
+                .findInvitationByRegistrationToken(token);
+
+        ReadInvitation readInvitation;
+
+        switch (invitation.getType()){
+            case APARTMENT:
+                apartmentInvitationService.registerWithRegistrationToken(token);
+                readInvitation = mapper.convert(invitation, ReadApartmentInvitation.class);
+                break;
+            case COOPERATION:
+                cooperationInvitationService.registerWithRegistrationToken(token);
+                readInvitation = mapper.convert(invitation, ReadCooperationInvitation.class);
+                break;
```

Comment:
Where is the expiration of the invitation for the coopAdmin checked?

Reply:
Added check to InvitationServiceImpl

Reply:
Sorry, but I can't find where we check if "LocalDateTime requestEndTime" expired in CooperationInvitation.

Reply:
We don't have to, we just need to check status, because we have job which deactivated invitations if they are expired, so we are delegating checking DateTime to it